tests/test_types_typeddict.py

```diff
@@ -470,15 +467,14 @@ def module():
 
         T = TypeVar('T')
 
+        class RecursiveGenTypedDictModel(BaseModel, Generic[T]):
+            rec: 'RecursiveGenTypedDict[T]'
+            model_config = dict(undefined_types_warning=False)
+
         class RecursiveGenTypedDict(TypedDict, Generic[T]):
-            # TODO: See if we can get this working if defined in a function (right now, needs to be module-level)
```

Comment:
I'm proud I was able to get this working — this was a matter of passing the global/local namespaces in the right places